### PR TITLE
Completed Capstone React Project

### PR DESCRIPTION
**Pull request by AsimKhan2019**
Completed the Capstone React Project for review. 

Completed the Home Page requirement
Completed the Details Page requirement 
Completed the Testing requirement 
Completed the Technical requirements for SPA and stateless calls and actions. 

Passing in the ID for Details and only calling the API if the data in the details category is empty or null as you can see and check. The search filter was not a requirement so I did not implement it. 

My Loom video presentation 

https://www.loom.com/share/27f4a4f105d5409cabc6b5786f897d0f
https://www.loom.com/share/5c1c68f39bf145d2b4407885603fbf2e

### CODE REVIEW

File: src/featureSlice/population/populationSlice.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      // console.log(resp.data);
+      resp.data.shift();
+      const s = resp.data.map((a) => ({
+        Id: a[2],
+        State: a[0],
+        Population: a[1],
+        Counties: [],
+      })).sort((a, b) => (parseInt(a.Population, 10) > parseInt(b.Population, 10) ? -1 : 0));
+      // return resp.data;
+      // console.log(s);
```
Comment (Wes-Isaac) [inline src/featureSlice/population/populationSlice.js]: - [x] Kindly remove commented code to keep the flow of your code.

File: README.md [hunk 2]
```diff
@@ -1,70 +1,90 @@
...
+- Git Flow, Webpack, Jest
+- React JS
+- React Redux 
+- React Logger
+- React Router
 
-The page will reload when you make changes.\
-You may also see any lint errors in the console.
+## Live Demo (if available)
+[Live Preview](https://6311cafbdda003000ae2de4b--glistening-biscotti-7c4251.netlify.app/)
```
Comment (Wes-Isaac) [inline README.md]: - [x] Kindly add your presentation link just like you did for the live link, and in the video please mention the part in the project that you are most proud of.
Screen Shot 2022-09-02 at 1 02 37 PM

File: src/reportWebVitals.js [hunk 3]
```diff
@@ -1,6 +1,8 @@
-const reportWebVitals = onPerfEntry => {
+const reportWebVitals = (onPerfEntry) => {
   if (onPerfEntry && onPerfEntry instanceof Function) {
-    import('web-vitals').then(({ getCLS, getFID, getFCP, getLCP, getTTFB }) => {
+    import('web-vitals').then(({
+      getCLS, getFID, getFCP, getLCP, getTTFB,
+    }) => {
```
Comment (Wes-Isaac) [inline src/reportWebVitals.js]: - [x] Kindly remove unused files that are generated while running `create-react-app` as it makes your code unclean.

**Review by Wes-Isaac — CHANGES_REQUESTED** [review]
Hi @AsimKhan2019, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!


## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
-N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/featureSlice/population/populationSlice.js (hunk 1 above)
Comment (AsimKhan2019) [reply src/featureSlice/population/populationSlice.js]: I am so sorry, I missed the important thing.

File: src/reportWebVitals.js (hunk 3 above)
Comment (AsimKhan2019) [reply src/reportWebVitals.js]: Ok I will delete all the unnecessary files

File: src/featureSlice/population/populationSlice.js (hunk 1 above)
Comment (Wes-Isaac) [reply src/featureSlice/population/populationSlice.js]: You did great, minor issues

File: README.md (hunk 2 above)
Comment (AsimKhan2019) [reply README.md]: I will make another video to display the implementation on netlify.

Comment (AsimKhan2019) [reply README.md]: https://www.loom.com/share/5c1c68f39bf145d2b4407885603fbf2e

File: src/featureSlice/population/populationSlice.js (hunk 1 above)
Comment (AsimKhan2019) [reply src/featureSlice/population/populationSlice.js]: @Wes-Isaac I removed the commented code

File: README.md (hunk 2 above)
Comment (AsimKhan2019) [reply README.md]: @Wes-Isaac I made another loom video

File: src/reportWebVitals.js (hunk 3 above)
Comment (AsimKhan2019) [reply src/reportWebVitals.js]: @Wes-Isaac I deleted the unwanted files

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hello @AsimKhan2019, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Gitflow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

- [x] [**REQUIRED**] While you did a great work by implementing the changes required by the previous reviewer, It seems that you forget to design your webApp for the Desktop version. For now, your project is only responsive for the mobile version. Kindly, design it so that when we open it on a desktop or tablet device  all the empty(white) space(left and right) be filled with the countries' card population. you can do it by adding the number of columns for the countries' card.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Rolandm99 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AsimKhan2019** [comment]
@RolandM99 I have made the required changes, although it was not part of the project's requirement. 

description

design

The project description clearly specifies mobile web application design is to focus on the mobile metrics app. Furthermore, if you follow the website for design guidelines, it shows the design and display for a mobile app, not a desktop application. It is unclear what kind of desktop design is required for this project and is not mentioned in the project description. Please visit the site below for clarification. 

https://www.behance.net/gallery/31579789/Ballhead-App-(Free-PSDs)

The site does not give any guidelines for desktop design but focuses only on mobile design.

**Review by AnselemOdims — APPROVED** [review]
### Hi  @AsimKhan2019 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AsimKhan2019** [comment]
@AnselemOdims Thank you for the approval

**Pull request merged**